src/rulesets/Base/fastmath_able.jl

```diff
@@ -5,8 +5,33 @@ let
     # e.g. do not add `foo` unless `Base.FastMath.foo_fast` exists.
     fastable_ast = quote
         #  Trig-Basics
-        @scalar_rule cos(x) -(sin(x))
-        @scalar_rule sin(x) cos(x)
+        ## use `sincos` to comput sin and cos at the same time
+        ## for the rules for sin and cos
+        ## See issue: https://github.com/JuliaDiff/ChainRules.jl/issues/291s
+        ## sin
+        function rrule(::typeof(sin), x::Number)
+            sinx, cosx = sincos(x)
+            sin_pullback(Δy) = (NO_FIELDS, cosx * Δy)
```

Comment:
To be correct for complex numbers need to conjugate:
```suggestion
            sin_pullback(Δy) = (NO_FIELDS, cosx' * Δy)
```